src/api/analytics/gtag.js

```diff
@@ -0,0 +1,192 @@
+      newOptions ? `(new) ${format(newOptions)}` : ''
+    }`
+  })
+}
+
+const sharedDataset = (status) =>
+  event('shared_dataset', {
+    shared_state: status ? 'Processed' : 'Unprocessed'
+  })
+
+/* --- Links --- */
+const experimentPageClick = (from) =>
+  event('page_view', {
+    experiment_page_click_from: from
+  })
+
+const exploredUsageClick = (usage, type = 'dataset') =>
+  event(`click`, { [`${type}_explored_usage`]: usage })
+
+const navClick = (item, type = 'global') =>
+  event(`page_view`, {
+    [`${type}_nav_item`]: item
+  })
+
+const outboundClick = (url) => {
+  let parameter
+  switch (true) {
+    case url === links.alsf || url === links.ccdl:
+      parameter = 'alsf_domain_name'
+      break
+    case url === links.ccdl_donate:
+      parameter = 'donate_click_from'
+      break
+    case url.includes(links.refinebio_githubio):
+      parameter = 'alsf_githubio_click_from'
+      break
+    case url.includes(links.alsf_github):
+      parameter = 'alsf_repo_click_from'
+      break
+    case url.includes(links.refinebio_api_docs):
+      parameter = 'api_docs_click_from'
+      break
+    case url.includes(links.refinebio_docs):
+      parameter = 'docs_click_from'
+      break
+    case url.includes(links.ccdl_twitter):
+      parameter = 'social_media'
+      break
+    default:
+      break
+  }
+
+  event('click', {
+    [parameter]: url
+  })
+}
+
+/* --- Search --- */
+const filterCombination = (facets, query) => {
+  const filters = getSortedFilters(facets, query)
+  const combination = getFilterCombination(filters)
+
+  event('page_view', { filter_combination: combination })
+}
+
+const filterType = (type) => event('filter_type', { filter_type: type })
+
+const toggleFilterItem = (checked, item) =>
+  event('toggled_filter_item', {
+    toggled_filter_item: `${checked ? 'Add' : 'Remove'} - ${item}`
+  })
+
+const searchTerm = (term) => event('search_text', { search_text: term })
+
+// helpers
+const getFilterCombination = (sortedFilters) => {
+  const { keys, filters } = sortedFilters
+  let formattedFilters = ''
+
+  // due to GA char limit, keys names are abbreviated
+  for (const x of keys) {
+    if (filters[x]) {
+      formattedFilters = `${formattedFilters}[${
+        x === keys[0] ? 'O' : x[0].toUpperCase()
+      }]:${filters[x]} `
+    }
+  }
+
+  return formattedFilters
+}
+
+const getSortedFilters = (facets, query) => {
+  const sort = (filter) => {
+    const key = Object.keys(filter)[0]
+
+    return Array.isArray(filter[key])
+      ? filter[key]
+          .map((item) => formatFilterName(key, item))
+          .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()))
+      : formatFilterName(key, filter[key])
+  }
+
+  const [, o, p, t] = formatFacetNames(Object.keys(facets))
+  const filters = {}
+  for (const x of [o, p, t]) {
+    if (query[x]) {
+      filters[x] = sort({ [x]: query[x] })
+    }
+  }
+
+  return { keys: [o, p, t], filters }
+}
```

Comment:
I'm having some difficulty reading this. I think we should start by using some semantic variable names here.

I don't understand why we are dropping the first facet.

Reply:
The reason for removing the first facet, `has_publication`, from `facets` is that it's not included in the data collection for the most used filter combination.

```js
  const [h, o, p, t] = formatFacetNames(Object.keys(facets))  
// h = "has_publication" <- this is unused thus dropped
// o = "downloadable_organism"
// p = "platform"
// t = "technology"
 ```
Initially, I used the first letter of each facet item (with `o` assigned to `downloadable_organism`), but I've now renamed them to the following for better clarity and semantics:
- `o` to `organismFacet`
- `p` to `platformFacet`
- `t` to `technologyFacet`  

Additionally, `x` is now renamed to `item`. I’ve also added a comment explaining the reason for removing the first facet (at 506fe4c7c0577784a3afe532efd1ce11be048204). Let me know your thoughts on this, thank you, David!